src/nvim/ex_docmd.c

```diff
@@ -5995,7 +5995,14 @@ static void ex_quit(exarg_T *eap)
  */
 static void ex_cquit(exarg_T *eap)
 {
-  getout(1);
+  int exitval = eap->addr_count > 0 ? eap->line2 : 1;
+
+  // Exit status out of range.
+  if (exitval > 255) {
```

Comment:
I think that should too great exit values cause any issues the situation should be handled by `mch_exit` and not here and not even `getout`. AFAIK on linux it just works like a cast to `uint8_t`. BTW, it looks like according to the C99 standard `EXIT_FAILURE` is the only proper default: 0 or EXIT_SUCCESS indicates success (standard explicitly mentions both), EXIT_FAILURE indicates failure and everything else is “implementation-defined” with no particular meaning attached by the standard, but without claiming that any other value yields undefined behaviour either.

So do not wrap by yourself, and use EXIT_FAILURE. May need an addition to os/win_defs.h: import `stdlib.h` there, define EXIT_FAILURE to 1 if it is not already defined.

Reply:
```Patch
diff --git a/src/nvim/ex_docmd.c b/src/nvim/ex_docmd.c
index 6e7938046..d4ce78aa3 100644
--- a/src/nvim/ex_docmd.c
+++ b/src/nvim/ex_docmd.c
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <stdbool.h>
 #include <stdint.h>
+#include <stdlib.h>
 #include <inttypes.h>
 
 #include "nvim/vim.h"
@@ -5995,7 +5996,7 @@ static void ex_quit(exarg_T *eap)
  */
 static void ex_cquit(exarg_T *eap)
 {
-  getout(1);
+  getout(eap->addr_count > 0 ? (int)eap->line2 : EXIT_FAILURE);
 }
 
 /*
```

Reply:
The good thing is that Vim help was nice enough to not specify what “error code” is exactly, so EXIT_FAILURE will not go against the help even should it be not one (and I have it defined to 1).